packages/dds/merge-tree/src/client.ts

```diff
@@ -587,8 +603,18 @@ export class Client extends TypedEventEmitter<IClientEvents> {
 				clientArgs.clientId,
 			);
 		}
+		// eslint-disable-next-line import/no-deprecated
+		if (start !== undefined && (op as IMergeTreeObliterateMsg).before1 === false) {
```

Comment:
we should do `as` casting when its possible to avoid, you can check the type of the op, which will narrow the type